drivers/platform/xilinx/xilinx_spi_pl.c

```diff
@@ -326,10 +335,81 @@ static int32_t xil_spi_write_and_read_pl(struct spi_desc *desc, uint8_t *data,
 	return SUCCESS;
 }
 
+/* Read/write FIFOs to/from buffers and update it structure accordingly */
+static void inline _transfer_and_update(struct xspi_desc *xdesc,
```

Comment:
Please provide an argument why chosing this FIFO approach is better compared to a simple ARRAY approach

Reply:
It is easy to create the list (no need to handle array size and allocation).